platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/ConsensusTransaction.java

```diff
@@ -16,9 +16,16 @@
 
 package com.swirlds.platform.system.transaction;
 
-import com.swirlds.platform.system.ReachedConsensus;
+import java.time.Instant;
 
 /**
  * A transaction that has reached consensus.
  */
-public sealed interface ConsensusTransaction extends Transaction, ReachedConsensus permits ConsensusTransactionImpl {}
+public sealed interface ConsensusTransaction extends Transaction permits ConsensusTransactionImpl {
+    /**
+     * Returns the community's consensus timestamp for this item.
+     *
+     * @return the consensus timestamp
+     */
+    Instant getConsensusTimestamp();
```

Comment:
missing nullity annotation (also on implementations)

Reply:
This is a tricky one. The idea of this interface is that transactions are presented when they reach consensus through this interface. So technically this should be `@NonNull`. But the object that implements this interface also implements the interface of the pre-consensus transactions. Meaning the overriding method would be `@Nullable`. I say we leave this as is for the time being since its not new code, and maybe we can discuss and potentially get rid of one of these interfaces.